src/python/pants/bsp/util_rules/targets.py

```diff
@@ -163,18 +175,76 @@ async def bsp_dependency_sources(request: DependencySourcesParams) -> Dependency
 # -----------------------------------------------------------------------------------------------
 
 
+@union
+@dataclass(frozen=True)
+class BSPDependencyModulesFieldSet(FieldSet):
+    """FieldSet used to hook computing dependency modules."""
```

Comment:
Is the idea behind having a separate field set per capability to allow BSP backends to be added incrementally? Seems mostly reasonable, although there are probably some capabilities that could be merged together such that they needed to be implemented at once... because if you can't add thirdparty dependencies at all, the backend isn't likely to be useful, and so it would almost be better for an author to need to stub `@rules` rather than have lots of fieldsets to think about.

Reply:
Mainly because I have not yet considered what can and/or should be merged. I need to see how the code for Java and Scala backends ends up. It may be that they should be a common JVM rule, maybe not. For now, this will allow me to experiment on this in the Scala rules.